bot/main.js

```diff
@@ -0,0 +1,92 @@
+            switch (op[0]) {
+                case 0: case 1: t = op; break;
+                case 4: _.label++; return { value: op[1], done: false };
+                case 5: _.label++; y = op[1]; op = [0]; continue;
+                case 7: op = _.ops.pop(); _.trys.pop(); continue;
+                default:
+                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
+                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
+                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
+                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
+                    if (t[2]) _.ops.pop();
+                    _.trys.pop(); continue;
+            }
+            op = body.call(thisArg, _);
+        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
+        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
+    }
+};
+exports.__esModule = true;
+/**
+ * Wechaty - Conversational RPA SDK for Chatbot Makers.
+ *  - https://github.com/wechaty/wechaty
+ */
+var wechaty_1 = require("wechaty");
+var qrcode_terminal_1 = require("qrcode-terminal");
+// https://stackoverflow.com/a/42817956/1123955
+// https://github.com/motdotla/dotenv/issues/89#issuecomment-587753552
+require("dotenv/config.js");
+function onLogout(user) {
+    wechaty_1.log.info('StarterBot', '%s logout', user);
+}
+function onScan(qrcode, status) {
+    if (status === wechaty_1.ScanStatus.Waiting || status === wechaty_1.ScanStatus.Timeout) {
+        var qrcodeImageUrl = [
+            'https://wechaty.js.org/qrcode/',
+            encodeURIComponent(qrcode),
+        ].join('');
+        wechaty_1.log.info('StarterBot', 'onScan: %s(%s) - %s', wechaty_1.ScanStatus[status], status, qrcodeImageUrl);
+        qrcode_terminal_1["default"].generate(qrcode, { small: true }); // show qrcode on console
+    }
+    else {
+        wechaty_1.log.info('StarterBot', 'onScan: %s(%s)', wechaty_1.ScanStatus[status], status);
+    }
+}
+function onLogin(user) {
+    wechaty_1.log.info('StarterBot', '%s login', user);
+}
+function onMessage(msg) {
+    return __awaiter(this, void 0, void 0, function () {
+        return __generator(this, function (_a) {
+            switch (_a.label) {
+                case 0:
+                    wechaty_1.log.info('StarterBot', msg.toString());
+                    if (!(msg.text() === 'ding')) return [3 /*break*/, 2];
+                    return [4 /*yield*/, msg.say('dong')];
+                case 1:
+                    _a.sent();
+                    _a.label = 2;
+                case 2: return [2 /*return*/];
+            }
+        });
+    });
+}
+var bot = new wechaty_1.Wechaty({
+    name: 'ding-dong-bot'
+});
+bot.on('scan', onScan);
+bot.on('login', onLogin);
+bot.on('logout', onLogout);
+bot.on('message', onMessage);
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/philip-zhan/badminton-group/pullRequest?prid=8160404)
```suggestion
bot.on("message", onMessage);
```